upload/admin/controller/extension/module/smaily_for_opencart.php

```diff
@@ -440,6 +453,63 @@ public function index() {
         } else {
             $data['rss_limit'] = $this->config->get('module_smaily_for_opencart_rss_limit');
         }
+        // Abandoned Cart status table.
+        $this->load->model('extension/smailyforopencart/admin');
+        $arguments = '';
+
+        if (isset($this->request->get['sort'])) {
+            $sort = $this->request->get['sort'];
+        } else {
+            $sort = 'customer_id';
+        }
+
+        if (isset($this->request->get['order'])) {
+            $order = $this->request->get['order'];
+        } else {
+            $order = 'ASC';
+        }
+
+        if (isset($this->request->get['page'])) {
+            $page = $this->request->get['page'];
+        } else {
+            $page = 1;
+        }
+
+        if ($order === 'ASC') {
+            $arguments .= '&order=DESC';
+        } else {
+            $arguments .= '&order=ASC';
+        }
+
+        if (isset($this->request->get['page'])) {
+            $arguments .= '&page=' . $this->request->get['page'];
+        }
+
+        $data['sort_name'] = $this->url->link('extension/module/smaily_for_opencart', 'user_token=' . $this->session->data['user_token'] . '&sort=lastname' . $arguments, true);
+        $data['sort_email'] = $this->url->link('extension/module/smaily_for_opencart', 'user_token=' . $this->session->data['user_token'] . '&sort=email' . $arguments, true);
+        $data['sort_date'] = $this->url->link('extension/module/smaily_for_opencart', 'user_token=' . $this->session->data['user_token'] . '&sort=sent_time' . $arguments, true);
+        $data['sort_status'] = $this->url->link('extension/module/smaily_for_opencart', 'user_token=' . $this->session->data['user_token'] . '&sort=is_sent' . $arguments, true);
```

Comment:
`Url::link` method can accept second argument as an array and it can build the URL query arguments using `http_build_query` method for you. Please rework this to provide URL arguments using array.